File: taller-5-python-ronalcastro97/zooAnimales/animal.py

import zooAnimales

class Animal():
    _totalAnimales=0
    _zona= None
    def __init__(self,nombre,edad,habitat,genero):
        self._nombre=nombre
        self._edad=edad
        self._habitat=habitat
        self._genero=genero
        Animal._totalAnimales+=1
    
    def getNombre(self):
        return self._nombre
    
    def setNombre(self,nombre):
        self._nombre=nombre

    def getEdad(self):
        return self._edad
    
    def setEdad(self,edad):
        self._edad=edad

    def getHabitat(self):
        return self._habitat
    
    def setHabitat(self,habitat):
        self._habitat=habitat
    
    def getGenero(self):
        return self._genero
    def setGenero(self,genero):
        self._genero=genero
    
    @classmethod
    def getTotalAnimales(cls):
        return cls._totalAnimales
    
    @classmethod
    def setTotalAnimales(cls,totalAnimales):
        cls._totalAnimales=totalAnimales
    
    @classmethod
    def getZona(cls):
        return cls._zona
    @classmethod
    def setZona(cls,zona):
        cls._zona=zona

    def movimiento():
        return "desplazarse"
    
    @staticmethod
    def totalPorTipo():
        return "Mamiferos : " + str(zooAnimales.mamifero.Mamifero.cantidadMamiferos()) + "\nAves : " + str(zooAnimales.ave.Ave.cantidadAves()) + "\nReptiles : " + str(zooAnimales.reptil.Reptil.cantidadReptiles()) + "\nPeces : " + str(zooAnimales.pez.Pez.cantidadPeces()) + "\nAnfibios : " + str(zooAnimales.anfibio.Anfibio.cantidadAnfibios())

    def toString(self):
        if self._zona != None:
            return "Mi nombre es " + self._nombre + ", tengo una edad de " + str(self._edad) + ", habito en " + self._habitat + " y mi genero es " + self._genero + ", la zona en la que me ubico es " + self._zona.getNombre() + ", en el zoo " + self._zona.getZoo().getNombre()
        else:
            return "Mi nombre es " + self._nombre + ", tengo una edad de " + str(self._edad) + ", habito en " + self._habitat + " y mi genero es " + self._genero
        
    

File: taller-5-java-ronalcastro97/src/main/java/zooAnimales/Animal.java
package zooAnimales;
import gestion.*; 
import java.util.ArrayList;
import java.util.List;

public class Animal {
	public static int totalAnimales=0;
	private String nombre;
	private int edad;
	private String habitat;
	private String genero;
	private List<Zona> zona = new ArrayList<Zona>(1);
	
	public Animal(){
			Animal.totalAnimales++;
		}
	public Animal(String nombre, int edad, String habitat,String genero) {
		Animal.totalAnimales++;
		this.nombre=nombre;
		this.edad=edad;
		this.habitat=habitat;
		this.genero=genero;

		}
	public static int getTotalAnimales() {
		return totalAnimales;
	}
	public static void setTotalAnimales(int totalAnimales) {
		Animal.totalAnimales = totalAnimales;
	}
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public int getEdad() {
		return edad;
	}
	public void setEdad(int edad) {
		this.edad = edad;
	}
	public String getHabitat() {
		return habitat;
	}
	public void setHabitat(String habitat) {
		this.habitat = habitat;
	}
	public String getGenero() {
		return genero;
	}
	public void setGenero(String genero) {
		this.genero = genero;
	}
	public List<Zona> getZona() {
		return zona;
	} 
	public void setZona(List<Zona> zona) {
		this.zona = zona; 
	}
	public static int totalAnimales() {
		return Animal.totalAnimales;
	}
	public  static String  totalPorTipo() {
		 	int totalmamiferos = Mamifero.getListado().size();
			int totalaves = Ave.getListado().size();
			int totalreptiles = Reptil.getListado().size();
			int totalpeces = Pez.getListado().size();
			int totalanfibios = Anfibio.getListado().size();
			
			return ("Mamiferos: " + totalmamiferos + "\n" + 
			"Aves: " + totalaves + "\n" +
			"Reptiles: " + totalreptiles + "\n" +
			"Peces: " + totalpeces + "\n" +
			"Anfibios: " + totalanfibios);
    }
		 	
	@Override
	public String toString() { 
		if (this.zona.size() != 0) {
			
			return ("Mi nombre es " + this.nombre + 
					", tengo una edad de " + this.edad + 
					", habito en " + this.habitat + 
					" y mi genero es " + this.genero + 
					" la zona en la que me ubico es " + this.zona.get(0) + 
					", en el " + this.zona.get(0).getZoo());
			
		}
		
		else {
			
			return ("Mi nombre es " + this.nombre + 
					", tengo una edad de " + this.edad + 
					", habito en " + this.habitat + 
					" y mi genero es " + this.genero);
		}
        }

	public String  movimiento() {
		return "desplazarse";
	}
}

